File: Circle.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from geometry_msgs.msg import Twist

class drawcirclenode(Node):
    def __init__(self):
        super().__init__("Draw_circle")
        self.cmd_vel_pub_ = self.create_publisher(Twist,"/cmd_vel", 10)
        self.timer_ = self.create_timer(0.5,self.send_velocity)
        self.get_logger().info("Draw circle node has been started")

    def send_velocity(self):
        msg = Twist()
        msg.linear.x = 1.0
        msg.angular.z = 0.8
        self.cmd_vel_pub_.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node = drawcirclenode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__  == "__main__":
    main()

File: Turtlebot3_Navigation/Circle.cpp
#include <rclcpp/rclcpp.hpp>
#include <geometry_msgs/msg/twist.hpp>

class DrawCircleNode : public rclcpp::Node 
{
public:
    DrawCircleNode() : Node("draw_circle")
    {
        cmd_vel_pub_ = this->create_publisher<geometry_msgs::msg::Twist>("/cmd_vel", 10);
        timer_ = this->create_wall_timer(
            std::chrono::milliseconds(500), 
            std::bind(&DrawCircleNode::sendVelocity, this)
        );
        RCLCPP_INFO(this->get_logger(), "Draw circle node has been started");
    }

private:
    void sendVelocity()
    {
        auto msg = geometry_msgs::msg::Twist();
        msg.linear.x = 1.0;
        msg.angular.z = 0.8;
        cmd_vel_pub_->publish(msg);
    }

    rclcpp::Publisher<geometry_msgs::msg::Twist>::SharedPtr cmd_vel_pub_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char ** argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<DrawCircleNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}
